x-pack/legacy/plugins/uptime/public/components/functional/monitor_list/monitor_list_drawer/monitor_status_list.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext } from 'react';
+import { get } from 'lodash';
+import { EuiHealth, EuiSpacer, EuiCallOut } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { Check } from '../../../../../common/graphql/types';
+import { UptimeSettingsContext } from '../../../../contexts';
+import { LocationLink } from './location_link';
+
+interface MonitorStatusListProps {
+  /**
+   * Recent List of checks performed on monitor
+   */
+  checks: Check[];
+}
+
+export const MonitorStatusList = ({ checks }: MonitorStatusListProps) => {
+  const {
+    colors: { success, danger },
+  } = useContext(UptimeSettingsContext);
+
+  const upChecks: Set<string> = new Set();
+  const downChecks: Set<string> = new Set();
+
+  const UN_NAMED_LOCATION = 'unnamed-location';
+
+  checks.forEach((check: Check) => {
+    let location = get<string>(check, 'observer.geo.name', UN_NAMED_LOCATION);
+
+    if (location === null) {
+      location = UN_NAMED_LOCATION;
+    }
```

Comment:
Do the contents of this block ever run? Won't the type of `location` be `string | 'unnamed-location'`?

Reply:
if location isn't defined, it's being returned as null. Type should be improve as string | null. I will do that.

Reply:
👍